### PR TITLE
Pad channels to same length when loading audio

### PR DESCRIPTION
**Pull request by desh2608**
This PR addresses https://github.com/lhotse-speech/lhotse/issues/671

There is also a minor change to the `lhotse split` CLI which allows passing a `--no-pad` parameter to disable padding suffixes with 0 (useful when submitting jobs with qsub).

### CODE REVIEW

File: lhotse/audio.py [hunk 1]
```diff
@@ -591,8 +591,31 @@ def load_audio(
                 samples = np.delete(samples, channels_to_remove, axis=0)
             samples_per_source.append(samples)
 
-        # shape: (n_channels, n_samples)
-        audio = np.vstack(samples_per_source)
+        # There may be a mismatch in the number of samples between different channels. We
+        # check if the mismatch is within a reasonable tolerance and if so, we pad
+        # all channels to the length of the longest one.
+        allowed_diff = int(
+            ceil(LHOTSE_AUDIO_DURATION_MISMATCH_TOLERANCE * self.sampling_rate)
```
Comment (pzelasko) [inline lhotse/audio.py:598]:
```suggestion
            compute_num_samples(LHOTSE_AUDIO_DURATION_MISMATCH_TOLERANCE, sampling_rate=self.sampling_rate)
```

**Review by pzelasko — COMMENTED** [review]
Thanks, looks good to me, left one minor comment. Also, maybe this duration-checking code could be a separate function? `load_audio` is already quite lengthy as-is.

**Comment by desh2608** [comment]
> Thanks, looks good to me, left one minor comment. Also, maybe this duration-checking code could be a separate function? `load_audio` is already quite lengthy as-is.

Done.

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**